packages/react-server/src/ReactFlightServer.js

```diff
@@ -1105,14 +1110,14 @@ export function resolveModelToJSON(
     return escapeStringValue(value);
   }
 
-  if (
-    typeof value === 'boolean' ||
-    typeof value === 'number' ||
-    typeof value === 'undefined'
-  ) {
+  if (typeof value === 'boolean' || typeof value === 'number') {
     return value;
   }
 
+  if (typeof value === 'undefined') {
+    return serializeUndefined();
```

Comment:
This serializes properties on object that have the value undefined. This is technically more correct since it has the property but can we live with it being a missing property instead to save bytes?

Reply:
Checking for property existence with `in` is not uncommon in TypeScript since it plays well with type-narrowing. TypeScript also checks for correct usage of missing properties and properties that are `undefined` if [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) is enabled. 

Also not sure how people will feel about inconsistent props if being passed from server -> client vs server->serer and client->client.

Reply:
Maybe revisit this behind a flag once Server Components are more widely adopted?

Reply:
I think we'd have to start with the less correct behavior and then make it correct rather than the other way around. It'll be harder to break later.

Reply:
I'm ok saying that we just waste the bytes here though. You probably shouldn't be using so many optional undefined props anyway. Personally I always use `null` instead anyway.